### PR TITLE
DOM-20596: Add AKS compatible configuration options

### PR DESCRIPTION
**Pull request by Secretions**
* Outputs the AGW IP for use w/dns provisioning (or whatever)
* Allows input of raw IP address(es) for the backend pool (ie for things not yet provisioned like a load balancer that will be setup w/a reserved IP later)
* Allows input of SSL cert in PFX format
* Requires password for SSL cert, as azurerm providers from 2.2.x+ fail if the password field is blank
* Remove self-signed SSL, as the above breaks it as-is

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -32,19 +28,24 @@ variable "targets" {
...
   description = "Enable the HTTP/2 protocol"
   default     = true
 }
 
-variable "ssl_cn" {
-  description = "Common name (CN) added to the self-signed certificate subject"
-  default     = "*.dominodatalab.com"
+variable "ssl_cert_pfx_data" {
+  description = "Password-protected PFX format SSL cert to install into AGW"
+  default     = null
```
Comment (zs-ddl) [inline variables.tf:43]: Do you need to specify a `default = null` explicitly? Isn't that the natural default value?

File: outputs.tf [hunk 2]
```diff
@@ -12,3 +12,8 @@ output "public_ip" {
   description = "The public IP of the application gateway"
   value       = azurerm_public_ip.this.ip_address
 }
+
+output "agw_ip_id" {
+  description = "The id of the AGW ip"
+  value       = azurerm_public_ip.this.id
```
Comment (zs-ddl) [inline outputs.tf:18]: Aw... I saw `this` used as the resource name in many examples. I'm still not sure how I feel about it. Just commenting, not suggesting  any changes be made.

Comment (Secretions) [reply outputs.tf:18]: I know how I feel about it, which is quite negatively. :) However, the convention (already in this module) was taken from existing conventions found in public terraform modules, ie:

https://github.com/terraform-aws-modules/terraform-aws-vpc/blob/master/main.tf

File: variables.tf (hunk 1 above)
Comment (Secretions) [reply variables.tf:43]: Should be, had these here when I was fighting to maintain the self-signed cert functionality I ultimately removed from the module which was no longer possible as-is due to newer azurerm providers requiring the password field (and thus breaking compatibility with non-password-protected pfx files). Ultimately gave up on that, so no reason not to nix these defaults (which didn't successfully kick it in the pants anyway).

**Pull request merged**